R/model_options.R

```diff
@@ -100,9 +100,32 @@ do_endpoint_model_options <- function(shape, survey, programme, anc) {
     anc_art_coverage_year2 = anc_year2_default
   )
 
-  naomi.options::get_controls_json("model", iso3, options, values,
-                                   config = list(include_art = has_art,
-                                                 include_anc = has_anc))
+  additional_control_groups <- NULL
+  if (use_mock_model()) {
+    additional_control_groups <- list(
+      list(
+        label = "Trigger mock model error",
+        controls = list(
+          list(
+            name = "mock_model_trigger_error",
+            type = "select",
+            help_text = "Set TRUE to force the model fit to error",
+            required = TRUE,
+            options = list(list(id = "true", label = "Yes"),
+                           list(id = "false", label = "No")),
+            value = "false"
```

Comment:
I'm surprised that this uses `"false"` here for FALSE, but presume that's correct. What's the serialisation approach?

Reply:
Yeah so for this "select control" type the values all have to be a string see https://github.com/mrc-ide/hintr/blob/master/inst/schema/ModelRunOptions.schema.json#L11 . It would be nice if we could have boolean types but would have to be updated in dynamic form in front end and we just went with doing the conversion from string to boolean here instead